File: src/test/java/com/twsg/lnl/kotlin/c3_extensions/g1_java/StringUtil.java
package com.twsg.lnl.kotlin.c3_extensions.g1_java;

import java.util.stream.IntStream;

import static com.twsg.lnl.kotlin.c3_extensions.g1_java.CharacterUtil.isVowel;

/**
 * @author tonyl
 */
public class StringUtil {

    private StringUtil() {
    }

    public static int countVowels(String value) {
        IntStream characters = value.chars();
        return (int) characters
                .filter(character ->
                        isVowel((char) character)
                )
                .count();
    }


}

File: src/test/java/com/twsg/lnl/kotlin/c3_extensions/g2_kotlin/StringUtil.kt
package com.twsg.lnl.kotlin.c3_extensions.g2_kotlin

/**
 * @author tonyl
 */

/* extension method */

fun String.countVowels(): Int {
    return this.asSequence()
            .filter(Char::isVowel)
            .count()
}
